core/src/main/scala/com/broadinstitute/dsp/models.scala

```diff
@@ -12,4 +13,14 @@ object CloudService {
 final case class Disk(id: Long, googleProject: GoogleProject, diskName: DiskName) {
   override def toString: String = s"${id}/${googleProject.value},${diskName.value}"
 }
+
+//init buckets are different than staging buckets because we store them with gs://[GcsBucketName]/
+case class InitBucketName(value: String) extends ValueObject with Product with Serializable {
```

Comment:
Thats true... I'lll move the parsing logic to the case class instead of the decoder. I looked at it initially like a sanitization issue, and we usually sanitize/check data from external sources before wrapping them in a case class, but we can do it the other way too. The gs:// string is never really used though

Reply:
since we're treating this as a separate type, so I'm seeing `InitBucketName` as the type for buckets with `gs://` prefix...but maybe we don't need a custom type at all if you're doing decoding when we're reading from DB...where's that code btw, I didn't see it

Reply:
oh nvm..it's right up there....yea, I think we do need a custom type...I kinda prefer to have `InitBucketName` having `gs://` there...I still like having the decoding to check if this is valid tho 🤔

Reply:
I compromised by making the case class enforce the prefix `gs://` and suffice `/` at construction